src/HealthGPS.Console/model_parser.cpp

```diff
@@ -141,59 +141,107 @@ std::unique_ptr<hgps::StaticLinearModelDefinition>
+        policy_model.coefficients = policy_json_params["Coefficients"]
+                                        .get<std::unordered_map<hgps::core::Identifier, double>>();
+        policy_model.log_coefficients =
+            policy_json_params["LogCoefficients"]
+                .get<std::unordered_map<hgps::core::Identifier, double>>();
+
+        // Check intervention policy covariance matrix column name matches risk factor name.
+        auto policy_column_name = policy_covariance_table.column(i).name();
+        if (!hgps::core::case_insensitive::equals(key, policy_column_name)) {
+            throw hgps::core::HgpsException{
+                fmt::format("Risk factor {} name ({}) does not match intervention "
+                            "policy covariance matrix column {} name ({})",
+                            i, key, i, policy_column_name)};
+        }
+
+        // Write intervention policy data structures.
+        policy_models.emplace_back(std::move(policy_model));
+        policy_ranges.emplace_back(policy_json_params["Range"].get<hgps::core::DoubleInterval>());
+        for (size_t j = 0; j < policy_covariance_table.num_rows(); j++) {
+            policy_covariance(i, j) =
+                std::any_cast<double>(policy_covariance_table.column(i).value(j));
         }
 
         // Increment table column index.
         i++;
     }
 
-    // Check correlation matrix column count matches risk factor count.
-    if (opt["RiskFactorModels"].size() != correlations_table.num_columns()) {
+    // Check risk factor correlation matrix column count matches risk factor count.
+    if (opt["RiskFactorModels"].size() != correlation_table.num_columns()) {
+        throw hgps::core::HgpsException{fmt::format("Risk factor count ({}) does not match risk "
+                                                    "factor correlation matrix column count ({})",
+                                                    opt["RiskFactorModels"].size(),
+                                                    correlation_table.num_columns())};
+    }
+
+    // Compute Cholesky decomposition of the risk factor correlation matrix.
+    auto cholesky = Eigen::MatrixXd{Eigen::LLT<Eigen::MatrixXd>{correlation}.matrixL()};
+
+    // Check intervention policy covariance matrix column count matches risk factor count.
+    if (opt["RiskFactorModels"].size() != policy_covariance_table.num_columns()) {
         throw hgps::core::HgpsException{
-            fmt::format("Risk factor count ({}) does not match correlation "
-                        "matrix column count ({})",
-                        opt["RiskFactorModels"].size(), correlations_table.num_columns())};
+            fmt::format("Risk factor count ({}) does not match intervention "
+                        "policy covariance matrix column count ({})",
+                        opt["RiskFactorModels"].size(), policy_covariance_table.num_columns())};
     }
```

Comment:
So, out of curiosity, what should happen if a particular risk factor does not have any intervention in the current policy? In other words, if it should be left "untouched".

Reply:
Presumably bad things... The workaround is to give a policy intercept of zero, with an empty coefficient array, but I agree that's not ideal. Once #231 is done, ideally intervention should become its own module, or just an instance of the static linear model. Maybe for now I can just autocomplete a no-op policy model, if no policy model is given. TLDR, bloated, needs refactoring.